src/stores/BibliographyStore.js

```diff
@@ -87,12 +88,17 @@ const BibliographyStoreModel = types
       }));
     },
     addCitation(newCitation) {
-      self.citations.push(newCitation);
-      self.latestId = newCitation.id;
-      localStorage.setItem(self.name, JSON.stringify({
-        style: self.style,
-        citations: self.citations,
-      }));
+      const isDuplicate = (self.citations.filter(citation => citation.url === newCitation.url).length > 0) ? true : false;
```

Comment:
ternary is unnecessary here
also could just replace `isDuplicate` in the if condition with `self.citations.filter(citation => citation.url === newCitation.url).length > 0`

Reply:
good point